src/scenic/simulators/carla/utils/utils.py

```diff
@@ -25,11 +26,13 @@ def scenicToCarlaVector3D(x, y, z=0.0):
     return carla.Vector3D(x, -y, z)
 
 
-def scenicToCarlaLocation(pos, z=None, world=None, blueprint=None):
-    if z is None:
+def scenicToCarlaLocation(pos, world=None, blueprint=None, snapToGround=False):
+    if snapToGround:
         assert world is not None
-        return snapToGround(world, carla.Location(pos.x, -pos.y, 0.0), blueprint)
-    return carla.Location(pos.x, -pos.y, z)
+        return scenic.simulators.carla.utils.utils.snapToGround(
```

Comment:
It's strange to use a fully-qualified name for this purely internal function. Why not rename the function to `_snapToGround`? Alternatively you could make it a pure function and call it something like `_groundSpawnPointFor`.